mountpoint-s3/tests/common/mod.rs

```diff
@@ -97,3 +102,36 @@ fn init_tracing_subscriber() {
     let _ = RustLogAdapter::try_init();
     let _ = tracing_subscriber::fmt::try_init();
 }
+
+#[cfg(feature = "s3_tests")]
+pub async fn get_test_sdk_sts_client() -> aws_sdk_sts::Client {
+    let config = aws_config::from_env()
+        .region(Region::new(get_test_region()))
+        .load()
+        .await;
+    aws_sdk_sts::Client::new(&config)
+}
+
+#[cfg(feature = "s3_tests")]
+pub async fn get_scoped_down_credentials(policy: &str) -> Credentials {
+    let sts_client = get_test_sdk_sts_client().await;
+    let assume_role_response = sts_client
+        .assume_role()
+        .role_arn(get_subsession_iam_role())
+        .role_session_name("mountpoint-s3-client_tests")
+        .policy(policy)
+        .send()
+        .await
+        .expect("assume_role with valid ARN and policy should succeed");
+    let credentials = assume_role_response
+        .credentials()
+        .expect("credentials should be present if assume_role succeeded")
+        .to_owned();
+    Credentials::new(
+        credentials.access_key_id().unwrap(),
+        credentials.secret_access_key().unwrap(),
+        credentials.session_token().map(|s| s.to_owned()),
+        None,
+        "scoped_down_sts_creds",
+    )
```

Comment:
I really should get #684 merged.

We need to redact these credentials if you're going to use them with `Command`, since it prints out the environment on failure.

For example:

https://github.com/awslabs/mountpoint-s3/blob/16cb2ed1baf218083aa48e96c1059c53e6983680/mountpoint-s3-client/tests/common/creds.rs#L25-L38

Reply:
to make it work we will need to add `--nocapture` [here](https://github.com/awslabs/mountpoint-s3/blob/main/.github/workflows/integration.yml#L98), right?

Reply:
That's a good thing to question...

I believe that it shouldn't matter. If the test fails, then the captured output should then be flushed to stdout and the masking pattern should be processed.

I wonder if it matters if Cargo switches it to `stderr`...

Reply:
Not much about output streams in the doc: https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#masking-a-value-in-a-log

To be completely safe we can print this both to stderr and stdout; thus we are making sure that any other occurrence of creds in either of the streams (e.g. because of panic) will come after a masking line `::add-mask::..`. 

Currently, it *may* happen, depending on github's runner implementation, that an unmasked value in stderr will be processed earlier than the `::add-mask::..` line in stdout

Reply:
I am leaving this as it is, only printing to `stdout`.

First of all, currently, all output of the test (even of `eprintln!()`) goes to `stdout` of cargo process, e.g:
```
# looking at stderr only
$ cargo test -- test_cmd_env 1>/dev/null
    Finished test [unoptimized + debuginfo] target(s) in 0.30s
     Running unittests src/lib.rs (target/debug/deps/fuser-7c62d545ed9ded77)
     Running tests/integration_tests.rs (target/debug/deps/integration_tests-c4309ec71a67c1eb)
     Running unittests src/lib.rs (target/debug/deps/mountpoint_s3-621df098ed460e0c)
     Running unittests src/main.rs (target/debug/deps/mount_s3-04d2109b3c02fbc6)
     Running tests/cli.rs (target/debug/deps/cli-74459daf07d48e44)
error: test failed, to rerun pass `-p mountpoint-s3 --test cli`
```

```
# looking at stdout only
$ cargo test -- test_cmd_env 2>/dev/null
<...>
failures:

---- test_cmd_env stdout ----
::add-mask::extraodinary_secret_value_stdout
::add-mask::extraodinary_secret_value_stderr
printing this once again: extraodinary_secret_value_stdout
printing this once again: extraodinary_secret_value_stderr
thread 'test_cmd_env' panicked at mountpoint-s3/tests/cli.rs:222:5:
fail the test
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    test_cmd_env

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 14 filtered out; finished in 0.00s
```

There were a couple of suggestions of printing that to stderr, but all closed with explanation that "machine-redable output will always go to stdout", output of the test falls into that category:
https://github.com/actions/runner/issues/891
https://github.com/rust-lang/cargo/issues/5641